### PR TITLE
feat!: migrate to use microgen

### PR DESCRIPTION
**Pull request by arithmetic1728**
Handwritten layer changes:
(1) `google/cloud/vision_helpers/__init__.py`: https://github.com/googleapis/python-vision/pull/52/files#diff-0fe929b13ddc6cf4577dc0d6f0048519
(2) `google/cloud/vision_helpers/decorators.py`: https://github.com/googleapis/python-vision/pull/52/files#diff-35a1f6af0c72e84a1bfb2c4f32b1a7b9
(3) `synth.py`

### CODE REVIEW

File: google/cloud/vision_helpers/__init__.py [hunk 1]
```diff
@@ -26,7 +27,7 @@ class VisionHelpers(object):
     See the :class:`~google.cloud.vision_v1.ImageAnnotatorClient`.
     """
 
-    def annotate_image(self, request, retry=None, timeout=None):
+    def annotate_image(self, request, retry=None, timeout=None, metadata=()):
```
Comment (software-dov) [inline google/cloud/vision_helpers/__init__.py:30]: I think that retry, timeout, and metadata should be keyword only to maintain consistency with the generated surface. Other people may disagree, and their opinions are probably more relevant.

File: google/cloud/vision_helpers/__init__.py [hunk 2]
```diff
@@ -40,36 +41,42 @@ def annotate_image(self, request, retry=None, timeout=None):
...
-        # If a filename is provided, read the file.
-        filename = protobuf.get(image, "source.filename", default=None)
-        if filename:
-            with io.open(filename, "rb") as img_file:
-                protobuf.set(request, "image.content", img_file.read())
-                protobuf.set(request, "image.source", None)
+                # If a filename is provided, read the file.
+                filename = protobuf.get(image, "source.filename", default=None)
+                if filename:
+                    with io.open(filename, "rb") as img_file:
```
Comment (software-dov) [inline google/cloud/vision_helpers/__init__.py:67]: Nit: don't need `io.open` anymore; just use `with open(filename, "rb") as img_file`

File: google/cloud/vision_helpers/__init__.py (hunk 1 above)
Comment (software-dov) [inline google/cloud/vision_helpers/__init__.py:30]: The new logic for this method doesn't feel quite right. I'll do another, more in depth pass later today, but I think that the type checking logic hides some potentially confusing new behavior.

File: google/cloud/vision_helpers/decorators.py [hunk 3]
```diff
@@ -71,24 +71,28 @@ def _create_single_feature_method(feature):
...
     """
 
     # Get the actual feature value to send.
-    feature_value = {"type": feature}
+    feature_value = {"type_": feature}
 
     # Define the function to be returned.
-    def inner(self, image, max_results=None, retry=None, timeout=None, **kwargs):
+    def inner(
+        self, image, max_results=None, retry=None, timeout=None, metadata=(), **kwargs
```
Comment (software-dov) [inline google/cloud/vision_helpers/decorators.py:94]: I think retry, timeout, and metadata should also be keyword only arguments for consistency.

File: google/cloud/vision_helpers/__init__.py (hunk 1 above)
Comment (busunkim96) [reply google/cloud/vision_helpers/__init__.py:30]: +1 to this. @telpirion or @sirtorry could you also weigh in? New generated clients look like `def some_method(self, request, *, retry=None, timeout=None, metadata=())`. (request positional, everything else keyword only)

Comment (arithmetic1728) [reply google/cloud/vision_helpers/__init__.py:30]: Line 104, `request = dict(image=image, features=[copied_features], **kwargs)`. All these keywords will be part of the request obejct, so `retry`, `timeout` etc are not in `kwargs`.

File: google/cloud/vision_helpers/decorators.py (hunk 3 above)
Comment (arithmetic1728) [reply google/cloud/vision_helpers/decorators.py:94]: https://github.com/googleapis/python-vision/blob/sijun/google/cloud/vision_helpers/decorators.py#L104

Comment (software-dov) [reply google/cloud/vision_helpers/decorators.py:94]: I don't think that's a problem. If the signature is something like
```python
def inner(self, image, *, max_results=None, retry=None, timeout=None, metadata=(), **kwargs):
   ...
```
Then `kwargs` doesn't accidentally capture retry/timeout/metadata.
I think `max_results` should also be keyword only, but I'm less confidant about that.

File: google/cloud/vision_helpers/__init__.py (hunk 2 above)
Comment (arithmetic1728) [reply google/cloud/vision_helpers/__init__.py]: done

File: google/cloud/vision_helpers/__init__.py (hunk 1 above)
Comment (arithmetic1728) [reply google/cloud/vision_helpers/__init__.py]: changed it to `def some_method(self, request, *, retry=None, timeout=None, metadata=())`

File: google/cloud/vision_helpers/decorators.py (hunk 3 above)
Comment (arithmetic1728) [reply google/cloud/vision_helpers/decorators.py]: changed it to `def inner(self, image, *, max_results=None, retry=None, timeout=None, metadata=(), **kwargs)`

File: google/cloud/vision_helpers/__init__.py (hunk 1 above)
Comment (arithmetic1728) [reply google/cloud/vision_helpers/__init__.py]: The original code supports `dict` and the classes defined in [`image_annotator_pb2.py`](https://github.com/googleapis/python-vision/blob/master/google/cloud/vision_v1/proto/image_annotator_pb2.py), but not `proto.Message`. Also the two cases `# If the image is a file handler, set the content.` and `# If a filename is provided, read the file.` don't look like relevant to `proto.Message`. I did some tweaks to make it also work for `proto.Message`.

**Review by busunkim96 — APPROVED** [review]

**Review by software-dov — APPROVED** [review]

**Comment by TurnrDev** [comment]
If this is a breaking change, what do I have to do to migrate my code? I swear this repo has the worst documentation.

**Comment by software-dov** [comment]
Take a look at `UPGRADING.md`

**Comment by busunkim96** [comment]
@TurnrDev There is a migration guide in the published reference documentation which is linked in the README under "Client Library Documentation". 

https://googleapis.dev/python/vision/latest/UPGRADING.html

**Pull request merged**